keras/engine/saving.py

```diff
@@ -244,73 +203,223 @@ def convert_custom_objects(obj):
+                              'optimizer.')
+
+    return model
+
+
+def save_model(model, filepath, overwrite=True, include_optimizer=True):
+    """Save a model to a HDF5 file.
+
+    Note: Please also see
+    [How can I install HDF5 or h5py to save my models in Keras?](
+        /getting-started/faq/
+        #how-can-i-install-HDF5-or-h5py-to-save-my-models-in-Keras)
+    in the FAQ for instructions on how to install `h5py`.
+
+    The saved model contains:
+        - the model's configuration (topology)
+        - the model's weights
+        - the model's optimizer's state (if any)
+
+    Thus the saved model can be reinstantiated in
+    the exact same state, without any of the code
+    used for model definition or training.
+
+    # Arguments
+        model: Keras model instance to be saved.
+        filepath: one of the following:
+            - string, path where to save the model, or
+            - h5py.File or h5py.Group object where to save the model
+        overwrite: Whether we should overwrite any existing
+            model at the target location, or instead
+            ask the user with a manual prompt.
+        include_optimizer: If True, save optimizer's state together.
+
+    # Raises
+        ImportError: if h5py is not available.
+    """
+    if h5py is None:
+        raise ImportError('`save_model` requires h5py.')
+
+    if not isinstance(filepath, h5py.Group):
+        # If file exists and should not be overwritten.
+        if not overwrite and os.path.isfile(filepath):
+            proceed = ask_to_proceed_with_overwrite(filepath)
+            if not proceed:
+                return
+        opened_new_file = True
+    else:
+        opened_new_file = False
+
+    f = h5dict(filepath, mode='w')
+
+    try:
+        _serialize_model(model, f, include_optimizer)
+    finally:
+        if opened_new_file:
+            f.close()
+
+
+def load_model(filepath, custom_objects=None, compile=True):
+    """Loads a model saved via `save_model`.
+
+    # Arguments
+        filepath: one of the following:
+            - string, path to the saved model, or
+            - h5py.File or h5py.Group object from which to load the model
+        custom_objects: Optional dictionary mapping names
+            (strings) to custom classes or functions to be
+            considered during deserialization.
+        compile: Boolean, whether to compile the model
+            after loading.
+
+    # Returns
+        A Keras model instance. If an optimizer was found
+        as part of the saved model, the model is already
+        compiled. Otherwise, the model is uncompiled and
+        a warning will be displayed. When `compile` is set
+        to False, the compilation is omitted without any
+        warning.
+
+    # Raises
+        ImportError: if h5py is not available.
+        ValueError: In case of an invalid savefile.
+    """
+    if h5py is None:
+        raise ImportError('`load_model` requires h5py.')
```

Comment:
This can be considered dead code since `h5py` is in the keras dependancies in the `setup.py`.

Reply:
It's good to have anyway. Keras could have been installed with`--no-deps`, etc.